### PR TITLE
Replaced Karma with Vitest in generated package

### PR DESCRIPTION
**Pull request by psmyrek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Other (generator): Replaced Karma with Vitest as the testing framework in the generated package.

Other (tools): Removed support for the `--coverage` (`-c`) and `--source-map` (`-s`) flags.

Other (tools): Removed `test` script from the tools, because unit tests in the generated package are executed now directly by Vitest.

MINOR BREAKING CHANGE (generator): The generated package no longer uses Karma as the test runner. Instead, Vitest is used.

MINOR BREAKING CHANGE (tools): The `test` script is removed, because unit tests in the generated package are executed directly by Vitest. Hence, the previous custom support for the `--coverage` (`-c`) and `--source-map` (`-s`) flags is no longer needed and has been also removed. The `yarn run test` (or `npm run test`) script is still available in the generated package, but it executes Vitest. See [CLI flags](https://vitest.dev/guide/cli.html) supported in Vitest.

---

### Additional information

Closes https://github.com/cksource/ckeditor5-internal/issues/3756.

### Supported scopes

* `generator` → https://www.npmjs.com/package/ckeditor5-package-generator
* `*` → https://www.npmjs.com/package/@ckeditor/ckeditor5-package-*

### CODE REVIEW

**Review by pomek — COMMENTED** [review]
@psmyrek, I think, we should disable watch by default. When a user wants to develop something, let's force passing the `--watch` modifier as it was implemented in Karma. WDYT?

**Comment by psmyrek** [comment]
Watch mode is now disabled in tests in the generated package. It can be enabled with `--watch` (`-w`) flag.

File: packages/ckeditor5-package-generator/lib/templates/js/vitest.config.ts.txt
Comment (pomek) [inline packages/ckeditor5-package-generator/lib/templates/js/vitest.config.ts.txt:1]: Shouldn't this file use the `*.js` extension? JavaScript packages => JS config, TS => TS.

Comment (psmyrek) [reply packages/ckeditor5-package-generator/lib/templates/js/vitest.config.ts.txt:1]: Of course, addressed in https://github.com/ckeditor/ckeditor5-package-generator/pull/187/commits/6e45f6c956af42bb406da8133c931fb40eb45e05.

**Review by pomek — APPROVED** [review]
LGTM.

**Comment by pomek** [comment]
@psmyrek, should we update CKEditor 5 docs?

* https://ckeditor.com/docs/ckeditor5/latest/framework/develpment-tools/package-generator/javascript-package.html
* https://ckeditor.com/docs/ckeditor5/latest/framework/develpment-tools/package-generator/typescript-package.html

**Comment by psmyrek** [comment]
PR with updates in the CKEditor 5 docs: https://github.com/ckeditor/ckeditor5/pull/17051

**Review by martnpaneq — APPROVED** [review]

**Review by pomek — APPROVED** [review]

**Pull request merged**